### PR TITLE
Custom ingest pipelines

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

- [x] Moves the custom ILM tutorial to its own sub-page of the Data Streams topic
- [x] Adds a custom ingest pipeline tutorial that follows the same outline as the custom ILM tutorial

Screen Shot 2022-08-16 at 10 52 58 PM

Preview the new tutorial by clicking **[here](https://observability-docs_2094.docs-preview.app.elstc.co/guide/en/fleet/master/data-streams-pipeline-tutorial.html)**.

### Related
* Closes https://github.com/elastic/observability-docs/issues/1936
* Closes https://github.com/elastic/observability-docs/issues/748
* Related to https://github.com/elastic/observability-docs/issues/1226

### CODE REVIEW

File: docs/en/ingest-management/data-streams.asciidoc [hunk 1]
```diff
@@ -240,3 +260,188 @@ or force a rollover using the {ref}/indices-rollover-index.html[{es} rollover AP
...
+
+. In {kib}, navigate to **Stack Management** -> **Ingest Pipelines** -> **Create pipeline** -> **New pipeline**.
+
+. Name your pipeline. We'll call this one, `add_field`.
+
+. Select **Add a processor**. Fill out the following information:
++
+** Processor: "Set"
+** Field: `test`
+** Value: `true`
```
Comment (bmorelli25) [inline docs/en/ingest-management/data-streams.asciidoc:290]: If anyone has a better idea for an example, I'm all ears.

**Comment by bmorelli25** [comment]
@nchaulet / @joshdover here's a first draft of custom ingest pipeline documentation. I could use your help to:

* make sure I'm on the right track here
* ensure there aren't any important details missing
* flesh out a better example if this one is too simple



@dedemorton you can hold off on a tech writer review for now

File: docs/en/ingest-management/data-streams.asciidoc [hunk 2]
```diff
@@ -108,9 +108,29 @@ By default, these data streams use an {ilm-init} policy that matches their data
 For example, the data stream `metrics-system.logs-*`,
 uses the metrics {ilm-init} policy as defined in the `metrics-system.logs` index template.
 
+Want to customize your index lifecycle management? See <<data-streams-ilm-tutorial>>.
+
 [discrete]
+[[data-streams-pipelines]]
+== Ingest pipelines
+
+{agent} integrations ship with a default {ref}/ingest.html[ingest pipeline]
```
Comment (nchaulet) [inline docs/en/ingest-management/data-streams.asciidoc:117]: Should we day `integrations datastreams ship with a default ...` as the relation is not 1 integration => 1 pipeline but 1 integration => n datastreams => 1 pipeline

Comment (bmorelli25) [reply docs/en/ingest-management/data-streams.asciidoc:117]: Good point 👍

**Comment by bmorelli25** [comment]
I need to merge this now, but I welcome any additional feedback and will address it in a follow-up PR. Thanks!

**Comment by felixbarny** [comment]
I think there's a leftover of outdated docs that could be confusing to users: https://www.elastic.co/guide/en/elasticsearch/reference/8.4/ingest.html#pipeline-custom-logs-configuration

Maybe we should replace this with a link to https://www.elastic.co/guide/en/fleet/current/data-streams-pipeline-tutorial.html?

But maybe there are some subtleties here. Not sure if the latter would work for the "custom logs" integration which doesn't have a default pipeline.

cc @joshdover

**Comment by bmorelli25** [comment]
Catching up after a long vacation. Thanks for pointing this out, Felix! I'll connect with the ES docs team to figure out the best path forward and get this fixed.

AFAIK, the latter should work with all integrations. @joshdover can you verify?

**Comment by nchaulet** [comment]
> @nchaulet can confirm, but my understanding is that it does work with custom logs

Just checked and it does not work with custom logs as we do not generate a correct index template for that stream

**Pull request merged**